### PR TITLE
Check Python test class namespace is prefixed with Test

### PR DESCRIPTION
**Pull request by codeinabox**
This addresses the bug reported in #232

### CODE REVIEW

**Comment by codeinabox** [comment]
@janko-m trying to figure out why this is happening, shouldn't `g:test#python#patterns` ensure that the test name must start with `test_`?

**Comment by janko-m** [comment]
@codeinabox The problem isn't that test.vim is not picking up the the `test_*` method, but that it's picking up the class that's inside the method, due to the `namespace` regex in `test#python#patterns`. 

I just took a look at "[Grouping multiple tests in a class](https://docs.pytest.org/en/latest/getting-started.html#grouping-multiple-tests-in-a-class)" and "[Conventions for Python test discovery](https://docs.pytest.org/en/latest/goodpractices.html#test-discovery)", and it seems that we're lucky, because names of classes that hold the `test_*` methods *need* to start with `Test`, so we just need to update the `namespace` regex with this constraint.

**Comment by codeinabox** [comment]
@janko-m thank you for spotting the test class naming convention. Have updated the PR accordingly.

@blueyed could you please test this branch against your tests.

**Comment by blueyed** [comment]
Will do.
However the Reeves should likely also handle unittest based tests: https://docs.pytest.org/en/latest/unittest.html

**Comment by codeinabox** [comment]
Good point. We should make the fixtures we have for Python are representative for the majority of cases

File: autoload/test/python.vim [hunk 1]
```diff
@@ -1,4 +1,4 @@
 let test#python#patterns = {
   \ 'test':      ['\v^\s*%(async )?def (test_\w+)'],
-  \ 'namespace': ['\v^\s*class (\w+)'],
+  \ 'namespace': ['\v^\s*class (Test\w+)'],
```
Comment (blueyed) [inline autoload/test/python.vim]: I think something like `'\v^\s*class .*Test'` could work, i.e. a class with `Test` in the name, or derived from something with `Test` in the name.
To be more strict `'\v^\s*class (Test\w+|.*TestCase.*)'` could be used maybe.

Comment (janko-m) [reply autoload/test/python.vim]: I agree with the stricter version 👍

**Comment by codeinabox** [comment]
@blueyed question regarding subclass tests, given the following test

```python
import unittest
import pytest

@pytest.mark.usefixtures("db_class")
class MyTest(unittest.TestCase):
    def test_method1(self):
        assert hasattr(self, "db")
        assert 0, self.db   # fail for demo purposes

    def test_method2(self):
        assert 0, self.db   # fail for demo purposes
```

If I do test nearest on line 7, should the expected command be `py.test test_subclass.py::MyTest::test_method1`?

**Comment by blueyed** [comment]
Yes.  Seems to work good.

**Comment by blueyed** [comment]
Assuming line 7 is `        assert hasattr(self, "db")`.

**Comment by janko-m** [comment]
Oops, I merged to soon, I didn't notice that we didn't add support for `unittest.TestCase` yet.

**Pull request merged**